src/Components/CreateImageWizardV2/steps/Repositories/Repositories.tsx

```diff
@@ -299,418 +337,221 @@ const Repositories = () => {
-    setSelected(selectedRepos);
-    updateFormState(selectedRepos);
-  };
-
-  const handleSelect = (
-    repoURL: string | undefined,
-    _: number,
-    isSelecting: boolean
-  ) => {
-    if (isSelecting === true) {
-      updateSelected([...selected, repoURL]);
-    } else if (isSelecting === false) {
-      if (repoURL && wizardMode === 'edit') {
-        setIsRepoRemovalModalOpen(true);
-        setRepoToRemove(repoURL);
-      } else {
-        updateSelected(
-          selected.filter((selectedRepoId) => selectedRepoId !== repoURL)
-        );
-      }
-    }
-  };
-
-  const handleSelectAll = () => {
-    if (data) {
-      updateSelected(
-        data.data
-          ?.filter(({ status }) => status === 'Valid')
-          .map((repo) => repo.url) || []
-      );
-    }
-  };
-
-  const computeStart = () => perPage * (page - 1);
-  const computeEnd = () => perPage * page;
-
-  const handleSelectPage = () => {
-    const pageRepos =
-      filteredRepositoryURLs &&
-      filteredRepositoryURLs.slice(computeStart(), computeEnd());
-
-    // Filter to avoid adding duplicates
-    const newSelected = pageRepos && [
-      ...pageRepos.filter((repoId) => !selected.includes(repoId)),
-    ];
-
-    updateSelected([...selected, ...newSelected!]);
+    setReposToRemove([]);
+    onClose();
   };
 
-  const handleDeselectAll = () => {
-    updateSelected([]);
-  };
-
-  const getRepoNameByUrl = (url: string) => {
-    return data!.data?.find((repo) => repo.url === url)?.name;
-  };
-
-  const handleCloseModalToggle = () => {
-    setIsRepoRemovalModalOpen(!isRepoRemovalModalOpen);
-  };
-
-  const handleConfirmRemovalModalToggle = () => {
-    updateSelected(
-      selected.filter((selectedRepoId) => selectedRepoId !== repoToRemove)
-    );
-    setIsRepoRemovalModalOpen(!isRepoRemovalModalOpen);
-  };
-
-  const RemoveRepositoryModal = () => {
-    return (
+  if (isError) return <Error />;
+  if (isLoading) return <Loading />;
+  return (
+    <Grid>
       <Modal
         titleIconVariant="warning"
-        title="Removing repositories can lead to build failures"
-        isOpen={isRepoRemovalModalOpen}
-        onClose={handleCloseModalToggle}
-        width="50%"
+        title="Are you sure?"
+        isOpen={modalOpen}
+        onClose={onClose}
+        variant="small"
         actions={[
-          <Button
-            key="remove"
-            variant="primary"
-            onClick={handleConfirmRemovalModalToggle}
-          >
+          <Button key="remove" variant="primary" onClick={handleRemoveAnyway}>
             Remove anyway
           </Button>,
-          <Button key="back" variant="link" onClick={handleCloseModalToggle}>
+          <Button key="back" variant="link" onClick={onClose}>
             Back
           </Button>,
         ]}
       >
-        We do not recommend removing any repositories if you have added packages
+        You are removing a previously added repository
```

Comment:
A teeny tiny nitpick.

```suggestion
        You are removing a previously added repository.
```